lib/msal-common/src/cache/entities/IdTokenEntity.ts

```diff
@@ -53,6 +57,27 @@ export class IdTokenEntity extends CredentialEntity {
     }
 
     /**
+     * fetches idToken from cache if present
+     * @param request
+     */
+    static readIdTokenFromCache(cacheManager: CacheManager, clientId: string, account: AccountInfo, realm?: string): IdTokenEntity {
+        if (StringUtils.isEmpty(realm)) {
+            const accountKey: string = AccountEntity.generateAccountCacheKey(account);
+            const cachedAccount = cacheManager.getAccount(accountKey);
+            realm = cachedAccount && cachedAccount.realm ? cachedAccount.realm : realm;
```

Comment:
Would prefer to not reassign `realm` (even if its empty).

Reply:
The `isEmpty` check is only needed for the `realm` value, the preceding two lines don't need to be guarded by it.

Reply:
the cached account is not needed if the realm is not empty. this saves a cache lookup.

Reply:
What if a helper function was used?

```js
const accountRealm = StringUtils.isEmpty(realm) ? this.getRealmForAccount(account) : realm;
```

Reply:
Done.